### PR TITLE
CORE-257 Add missing field "flags" to the Identity struct

### PR DESCRIPTION
**Pull request by ig-garcia**
I find this issue quite complex, please also check the comments in CORE-257 ticket. There could also be better solutions to the issue.

### CODE REVIEW

File: asn.1/message.asn1 [hunk 1]
```diff
@@ -53,7 +53,8 @@ PIdentity ::= SEQUENCE {
     user-id     PString,
     username    PString OPTIONAL,
     comm-type   INTEGER (0..255),
-    lang        ISO639-1 OPTIONAL
+    lang        ISO639-1 OPTIONAL,
+    flags       INTEGER (0..61440)
```
Comment (ghost) [inline asn.1/message.asn1]: 0xf000? Please use at least full 16 bits, that is 65535, easier to understand, and no risk of the asn1 code rejecting a valid pattern.

**Review by ghost — CHANGES_REQUESTED** [review]

File: asn.1/message.asn1 (hunk 1 above)
Comment (ig-garcia) [reply asn.1/message.asn1]: ok, I had just based it in the max possible value for the number of hexadecimal digits used.

File: test/src/KeyResetMessageTest.cc [hunk 2]
```diff
@@ -1481,6 +1487,8 @@ TEST_F(KeyResetMessageTest, check_reset_all_own_grouped_recv_with_sticky) {
 
 
 TEST_F(KeyResetMessageTest, check_reset_grouped_own_multiple_keys_multiple_idents_reset_all_recv) {
+    //CORE-257 test mails for this test case cannot be re-generated
```
Comment (ghost) [inline test/src/KeyResetMessageTest.cc]: Minor issue: Consistent spacing please, e.g. `// CORE-257 ...`.

File: test/src/KeyResetMessageTest.cc [hunk 3]
```diff
@@ -1707,6 +1715,8 @@ TEST_F(KeyResetMessageTest, check_reset_grouped_own_multiple_keys_multiple_ident
 }
 
 TEST_F(KeyResetMessageTest, check_reset_grouped_own_multiple_keys_multiple_idents_reset_one_recv) {
+    //CORE-257 test mails for this test case cannot be re-generated
```
Comment (ghost) [inline test/src/KeyResetMessageTest.cc]: Minor issue: Consistent spacing please, e.g. `// CORE-257 ...`.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
All in all solid code/changes, and tests run successfully. My only change requests are literally about adding a space in comments.

**Comment by ig-garcia** [comment]
I found that because of some changes we did in CORE-67 (leave device group on key reset), some test messages cannot be re-generated. So it is possible that the tests using those test messages do not make sense anymore either.

**Review by ghost — APPROVED** [review]

**Review by saschabacardit — APPROVED** [review]

**Pull request merged**